### PR TITLE
Add single-end SS2 pipeline

### PR DESCRIPTION
**Pull request by mckinsel**
### Purpose
Add a pipeline that processes single-end SS2 data.

Also, the SS2 paired end pipeline has a `gtf_file` input, but it doesn't use it. This removes that input from the adapter wdls, and there's a branch of the same name in skylab that removes it from the pipeline itself.

### CODE REVIEW

File: pipeline_tools/pipelines/smartseq2/smartseq2.py [hunk 1]
```diff
@@ -85,3 +85,87 @@ def _get_content_for_ss2_input_tsv(bundle_uuid, bundle_version, dss_url, http_re
     sample_id = metadata_utils.get_sample_id(primary_bundle)
     fastq_1_url, fastq_2_url = get_urls_to_files_for_ss2(primary_bundle)
     return fastq_1_url, fastq_2_url, sample_id
+
+
+def create_ss2_se_input_tsv(
+    bundle_uuid, bundle_version, dss_url, input_tsv_name='inputs.tsv'
+):
+    """Create TSV of Smart-seq2 inputs.
```
Comment (rexwangcc) [inline pipeline_tools/pipelines/smartseq2/smartseq2.py]:
```suggestion
    """Create TSV of Smart-seq2-SingleEnd inputs.
```
nit

File: adapter_pipelines/ss2_single_end/adapter.wdl [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  String reference_bundle
+
+  # Set runtime environment such as "dev" or "staging" or "prod" so submit task could choose proper docker image to use
+  String runtime_environment
+  # By default, don't record http requests, unless we override in inputs json
+  Boolean record_http = false
+  Int max_cromwell_retries = 0
+  Boolean add_md5s = false
+
+  String pipeline_tools_version = "v0.50.2"
```
Comment (rexwangcc) [inline adapter_pipelines/ss2_single_end/adapter.wdl]: It's annoying but until we separate the adapter WDLs from this repo, anything that changes the `pipeline-tools` package requires you to change this to a "pre-release version" for all of the WDLs. For instance, in this case, you'd change bump this to `v0.51.0` (even it's not tagged yet), and also change this line in all of the existing adapter WDLs.

File: pipeline_tools/pipelines/smartseq2/smartseq2.py [hunk 3]
```diff
@@ -85,3 +85,87 @@ def _get_content_for_ss2_input_tsv(bundle_uuid, bundle_version, dss_url, http_re
...
+        bundle (humancellatlas.data.metadata.Bundle): A Bundle object contains all of the necessary information.
+
+    Returns:
+         tuple: A tuple consisting of the url to the input fastq file
+         respectively.
+    """
+    fastq_url = None
+    sequence_files = bundle.sequencing_output
+
+    for sf in sequence_files:
```
Comment (samanehsan) [inline pipeline_tools/pipelines/smartseq2/smartseq2.py]: You can remove the check for `if fastq_url`, and move the return statement under `if sf.read_index == 'read1'` to simplify this a bit:
```
if sf.read_index == 'read1':
    return sf.manifest_entry.url
```

**Comment by samanehsan** [comment]
Looks like the travis tests are failing due to a flake8 error. The README has some info about how to easily reformat that: https://github.com/HumanCellAtlas/pipeline-tools#code-style

File: adapter_pipelines/ss2_single_end/adapter.wdl (hunk 2 above)
Comment (mckinsel) [reply adapter_pipelines/ss2_single_end/adapter.wdl]: Fixed

File: pipeline_tools/pipelines/smartseq2/smartseq2.py (hunk 3 above)
Comment (mckinsel) [reply pipeline_tools/pipelines/smartseq2/smartseq2.py]: Thanks, updated

File: adapter_pipelines/ss2_single_end/adapter_example_static.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "AdapterSmartSeq2SingleCellUnpaired.reference_bundle": "bf51d668-3e14-4843-9bc7-5d676fdf0e01",
+  "AdapterSmartSeq2SingleCellUnpaired.format_map": "gs://hca-dcp-mint-test-data/adapters/file_format_map.json",
+  "AdapterSmartSeq2SingleCellUnpaired.method": "SmartSeq2SingleCell",
+  "AdapterSmartSeq2SingleCellUnpaired.analysis_file_version": "6.0.0",
+  "AdapterSmartSeq2SingleCellUnpaired.analysis_protocol_schema_version": "9.0.0",
+  "AdapterSmartSeq2SingleCellUnpaired.analysis_process_schema_version": "11.0.1",
+  "AdapterSmartSeq2SingleCellUnpaired.run_type": "run",
+  "AdapterSmartSeq2SingleCellUnpaired.add_md5s": false,
+
+  "AdapterSmartSeq2SingleCellUnpaired.runtime_environment": "test",
```
Comment (samanehsan) [inline adapter_pipelines/ss2_single_end/adapter_example_static.json]: Please remove the inputs that were added in for testing prior to merging this PR :octocat:

**Comment by rexwangcc** [comment]
Shall we merge this PR (once the comment is addressed)? @mckinsel

File: adapter_pipelines/ss2_single_end/adapter_example_static.json (hunk 4 above)
Comment (rexwangcc) [reply adapter_pipelines/ss2_single_end/adapter_example_static.json]: removed :)

**Comment by rexwangcc** [comment]
Merging now. We can always make follow-up PRs later.

**Pull request merged**